modules/ballerina-mime/src/main/java/org/ballerinalang/mime/util/MimeUtil.java

```diff
@@ -0,0 +1,493 @@
+        BStruct fileStruct = ConnectorUtils
+                .createAndGetStruct(context, Constants.PROTOCOL_PACKAGE_FILE, Constants.FILE);
+        fileStruct.setStringField(TEMP_FILE_PATH_INDEX, temporaryFilePath);
+        entityStruct.setRefField(OVERFLOW_DATA_INDEX, fileStruct);
+        return entityStruct;
+    }
+
+    /**
+     * Given an input stream, create a temporary file and write the content on to it.
+     *
+     * @param inputStream Input stream coming from the request/response.
+     * @param fileName    Temporary file name
+     * @return Absolute path of the created temporary file.
+     */
+    private static String writeToTemporaryFile(InputStream inputStream, String fileName) {
+        OutputStream outputStream = null;
+        try {
+            File tempFile = File.createTempFile(fileName, TEMP_FILE_EXTENSION);
+            outputStream = new FileOutputStream(tempFile.getAbsolutePath());
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            //read from inputstream to buffer
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            inputStream.close();
+            //flush OutputStream to write any buffered data to file
+            outputStream.flush();
+            outputStream.close();
+            return tempFile.getAbsolutePath();
+        } catch (IOException e) {
+            throw new BallerinaException("Error while writing the payload info into a temp file: " + e.getMessage());
+        } finally {
+            try {
+                if (outputStream != null) {
+                    outputStream.close();
+                }
+            } catch (IOException e) {
+                LOG.error("Error occured while closing outputstream in writeToTemporaryFile", e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Read bytes from a given file.
+     *
+     * @param filePath a string representing the file
+     * @return bytes read from the given file
+     */
+    private static byte[] readFromFile(String filePath) {
+        try {
+            return Files.readAllBytes(Paths.get(filePath));
+        } catch (IOException e) {
+            throw new BallerinaException("Error while reading content from file handler: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Given an input stream, get a byte array.
+     *
+     * @param input Represent an input stream
+     * @return a byte array
+     * @throws IOException
+     */
+    public static byte[] toByteArray(InputStream input) throws IOException {
+        byte[] buffer = new byte[4096];
+        int n1;
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        for (; -1 != (n1 = input.read(buffer)); ) {
+            output.write(buffer, 0, n1);
+        }
+        byte[] bytes = output.toByteArray();
+        output.close();
+        return bytes;
+    }
```

Comment:
Can we simplify this method using try with resources like below which has less lines and lesser local variables?

`private static byte[] getByteArray(InputStream input) throws IOException { 

        try (ByteArrayOutputStream output = new ByteArrayOutputStream();) {
            byte[] buffer = new byte[4096];
            for (int len; (len = input.read(buffer)) != -1; ) {
                output.write(buffer, 0, len);
            }
            return output.toByteArray();
        }
    }`